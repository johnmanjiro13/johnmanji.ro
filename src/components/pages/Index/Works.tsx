import { Card } from "@/components/common/Card";
import { TitledSecton } from "@/components/common/TitledSection";

export function Works() {
  return (
    <TitledSecton title="Works">
      <div className="grid grid-cols-2 gap-4">
        <div className="row-span-3">
          <Card
            image={{
              src: "/images/gh-bump.jpg",
              width: 642,
              height: 276,
              alt: "gh-bump sample image",
            }}
            heading="gh-bump"
            content="An extension for gh cli which bumps version of a repository."
            url="https://github.com/johnmanjiro13/gh-bump"
          />
        </div>
        <div className="">
          <Card
            heading="gh-cmcm"
            content="A gh extension which comments to a commit by GitHub API. cmcm means 'commit comment'."
            url="https://github.com/johnmanjiro13/gh-cmcm"
          />
        </div>
        <div>
          <Card
            heading="tokio-fluent"
            content="A fluentd client for Rust using tokio."
            url="https://github.com/johnmanjiro13/tokio-fluent"
          />
        </div>
        <div>
          <Card
            heading="auto-reviewer-assign"
            content="A GitHub Action to assign reviewers to pull requests automatically."
            url="https://github.com/johnmanjiro13/auto-reviewer-assign"
          />
        </div>
      </div>
    </TitledSecton>
  );
}
