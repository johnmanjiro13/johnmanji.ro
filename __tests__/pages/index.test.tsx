import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import Index from "../../pages/index";

test("snapshot", () => {
  const { asFragment } = render(<Index />);
  expect(asFragment()).toMatchSnapshot();
});
