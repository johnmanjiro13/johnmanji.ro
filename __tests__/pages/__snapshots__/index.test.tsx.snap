// Vitest Snapshot v1

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen"
  >
    <main>
      <div
        class="relative mb-20"
      >
        <img
          alt="Background image for title header"
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Fheader.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Fheader.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fheader.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fheader.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fheader.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fheader.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fheader.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fheader.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fheader.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
        />
        <h1
          class="text-center text-5xl p-20 relative"
        >
          johnmanji.ro
        </h1>
      </div>
      <div
        class="container mx-auto px-4"
      >
        <section
          class="mb-20 mx-auto max-w-4xl"
        >
          <h2
            class="text-4xl mb-2 "
          >
            About me
          </h2>
          <hr
            class="mb-4 w-80"
          />
          <div
            class="flex flex-wrap"
          >
            <img
              alt="Ikumi Nakamura"
              class="rounded-full"
              data-nimg="1"
              decoding="async"
              height="144"
              src="/_next/image?url=%2Fimages%2Fprofile.jpg&w=384&q=75"
              srcset="/_next/image?url=%2Fimages%2Fprofile.jpg&w=256&q=75 1x, /_next/image?url=%2Fimages%2Fprofile.jpg&w=384&q=75 2x"
              style="color: transparent;"
              width="144"
            />
            <div
              class="ml-4 mt-4"
            >
              <h2
                class="text-2xl mb-2"
              >
                Ikumi Nakamura
              </h2>
              <p>
                I am a software engineer working in Tokyo, Japan and I am called Johnmanjiro.
                <br />
                I graduated Kurume NCT and transferred to Tsukuba university.
                <br />
                After dropping out of graduate school, I started working as a software engineer at Gunosy inc.
              </p>
            </div>
          </div>
        </section>
        <section
          class="mb-20 mx-auto max-w-4xl"
        >
          <h2
            class="text-4xl mb-2 "
          >
            Works
          </h2>
          <hr
            class="mb-4 w-80"
          />
          <div
            class="flex"
          >
            <div
              class="flex-initial"
            >
              <div
                class="max-w-sm rounded overflow-hidden shadow-lg relative m-2"
              >
                <div
                  class="px-6 py-4"
                >
                  <img
                    alt="gh-bump sample image"
                    data-nimg="1"
                    decoding="async"
                    height="276"
                    src="/_next/image?url=%2Fimages%2Fgh-bump.jpg&w=1920&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fgh-bump.jpg&w=750&q=75 1x, /_next/image?url=%2Fimages%2Fgh-bump.jpg&w=1920&q=75 2x"
                    style="color: transparent;"
                    width="642"
                  />
                  <div
                    class="font-bold text-xl mb-2"
                  >
                    gh-bump
                  </div>
                  <p
                    class="text-gray-700 text-base mb-2"
                  >
                    gh-bump is an extension for gh cli which bumps version of a repository.
                  </p>
                  <a
                    href="https://github.com/johnmanjiro13/gh-bump"
                  >
                    &gt;&gt; GitHub
                  </a>
                </div>
              </div>
            </div>
            <div
              class="flex-initial"
            >
              <div
                class="max-w-sm rounded overflow-hidden shadow-lg relative m-2"
              >
                <div
                  class="px-6 py-4"
                >
                  <div
                    class="font-bold text-xl mb-2"
                  >
                    gh-cmcm
                  </div>
                  <p
                    class="text-gray-700 text-base mb-2"
                  >
                    gh-cmcm (commit comment) is a gh extension which comments to a commit by GitHub API.
                  </p>
                  <a
                    href="https://github.com/johnmanjiro13/gh-cmcm"
                  >
                    &gt;&gt; GitHub
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="mb-20 mx-auto max-w-4xl"
        >
          <h2
            class="text-4xl mb-2 "
          >
            Employment history
          </h2>
          <hr
            class="mb-4 w-80"
          />
          <ul
            class="list-disc pl-4"
          >
            <li>
              Apr 2019 - current: software engineer at Gunosy Inc.
            </li>
          </ul>
        </section>
        <section
          class="mb-20 mx-auto max-w-4xl"
        >
          <h2
            class="text-4xl mb-2 "
          >
            Programming skills
          </h2>
          <hr
            class="mb-4 w-80"
          />
          <p
            class="mb-3"
          >
            I can use
          </p>
          <ul
            class="list-disc pl-6"
          >
            <li>
              Go
            </li>
            <li>
              Python
            </li>
            <li>
              Ruby / Ruby on Rails
            </li>
            <li>
              JavaScript / TypeScript
            </li>
            <li>
              React.js
            </li>
          </ul>
        </section>
        <section
          class="mb-20 mx-auto max-w-4xl"
        >
          <h2
            class="text-4xl mb-2 "
          >
            Contact
          </h2>
          <hr
            class="mb-4 w-80"
          />
          <div
            class="text-4xl"
          >
            <a
              href="https://github.com/johnmanjiro13"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-github "
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="ml-10"
              href="https://twitter.com/johnmanjiro13"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-twitter "
                data-icon="twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="ml-10"
              href="https://www.facebook.com/johnmanjiro13"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-facebook "
                data-icon="facebook"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              class="ml-10"
              href="https://johnmanjiro13.hatenablog.com/"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-blog "
                data-icon="blog"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M192 32c0 17.7 14.3 32 32 32c123.7 0 224 100.3 224 224c0 17.7 14.3 32 32 32s32-14.3 32-32C512 128.9 383.1 0 224 0c-17.7 0-32 14.3-32 32zm0 96c0 17.7 14.3 32 32 32c70.7 0 128 57.3 128 128c0 17.7 14.3 32 32 32s32-14.3 32-32c0-106-86-192-192-192c-17.7 0-32 14.3-32 32zM96 144c0-26.5-21.5-48-48-48S0 117.5 0 144V368c0 79.5 64.5 144 144 144s144-64.5 144-144s-64.5-144-144-144H128v96h16c26.5 0 48 21.5 48 48s-21.5 48-48 48s-48-21.5-48-48V144z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </section>
      </div>
    </main>
    <footer
      class="text-center mb-8"
    >
      Copyright © 2022- Ikumi Nakamura.
    </footer>
  </div>
</DocumentFragment>
`;
